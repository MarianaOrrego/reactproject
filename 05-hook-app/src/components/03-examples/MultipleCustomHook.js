import { useCouter } from '../../hooks/useCouter';
import { useFetch } from '../../hooks/useFetch';
import '../02-useEffect/effect.css';

export const MultipleCustomHook = () => {

    const {counter, increment} = useCouter(1);

    const {loading, data} = useFetch(`https://www.breakingbadapi.com/api/quotes/${ counter }`);

    const {author, quote } = !!data && data[0];

  return (
    <div>
      <h1>BreakingBad</h1>
      <hr />

      {
        loading
        ?
          (
            <div className='alert alert-info'>
                Loading...
            </div>
          )
        :
          (
            <blockquote className='blockquote text-right'>
              <p className='mb-0'>{ quote }</p>
              <footer className='blockquote-footer mt-1'>{ author }</footer>
            </blockquote>
          )
      }

      <div className='d-flex justify-content-center'>
        <button 
          className='btn btn-primary'
          onClick={ increment }>
          Siguiente frase
        </button>
      </div>

    </div>
  )
}