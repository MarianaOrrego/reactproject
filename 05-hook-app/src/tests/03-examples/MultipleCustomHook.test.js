import React from 'react';
import { shallow } from 'enzyme';
import { MultipleCustomHook } from '../../components/03-examples/MultipleCustomHook';
import { useFetch } from '../../hooks/useFetch';
import { useCouter } from '../../hooks/useCouter';

jest.mock('../../hooks/useFetch');
jest.mock('../../hooks/useCouter');




describe('Pruebas en <MultipleCustomHooks />', () => {

    beforeEach( () =>  {
 
        useCouter.mockReturnValue({
            counter: 10,
            increment: () => {}
        });
 
    });

    
    test('debe de mostrarse correctamente', () => {

        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        });

        const wrapper = shallow( <MultipleCustomHook /> );
        expect( wrapper ).toMatchSnapshot();
        
    })

    test('should debe de mostrar la informaciÃ³n', () => {
        
        useFetch.mockReturnValue({
            data: [{
                author: 'Mariana',
                quote: 'Pura vida'
            }],
            loading: false,
            error: null
        });

        const wrapper = shallow( <MultipleCustomHook /> );

        expect( wrapper.find('.alert').exists() ).toBe( false );
        expect( wrapper.find('.mb-0').text().trim() ).toBe( 'Pura vida' );
        expect( wrapper.find('footer').text().trim() ).toBe( 'Mariana' );

    })
    
    

})


