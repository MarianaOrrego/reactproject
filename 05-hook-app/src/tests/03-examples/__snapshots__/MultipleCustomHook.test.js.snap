// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MultipleCustomHooks /> debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    BreakingBad
  </h1>
  <hr />
  <div
    className="alert alert-info"
  >
    Loading...
  </div>
  <div
    className="d-flex justify-content-center"
  >
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Siguiente frase
    </button>
  </div>
</div>
`;
